friendshipper/src-tauri/src/repo/operations/revert.rs

```diff
@@ -39,20 +39,101 @@ where
             bail!("no files provided");
         }
 
+        let mut num_chars = 0;
+        for f in self.files.iter() {
+            num_chars += f.len();
+        }
+
         let branch = self.repo_status.read().branch.clone();
-        let mut args: Vec<&str> = vec!["checkout", &branch, "--"];
 
-        for file in &self.files {
-            args.push(file);
+        // windows command line length limit is 8191, so if we're close to that, checking using a file instead
+        if num_chars > 8000 {
+            match self.revert_with_listfile(&branch).await {
+                Ok(_) => return Ok(()),
+                Err(e) => {
+                    error!(
+                        "Revert files with listfile failed, falling back to chunked batches: {}",
+                        e
+                    )
+                }
+            }
         }
 
-        self.git_client.run(&args, git::Opts::default()).await?;
+        for chunk in self.files.chunks(50) {
```

Comment:
do you still wanna chunk even with the listfile bit?

Reply:
I was thinking if the listfile fails for any reason, it can fall back to the chunking code. But I suspect most of the time it won't need to chunk.

Reply:
Ah, okay, this is a fallback, cool